server/sonar-server/src/main/java/org/sonar/server/permission/ws/ProjectWsRef.java

```diff
@@ -39,7 +39,7 @@
   private ProjectWsRef(@Nullable String uuid, @Nullable String key) {
     this.uuid = uuid;
     this.key = key;
-    checkRequest(this.uuid != null ^ this.key != null, "Project id or project key can be provided, not both.");
+    checkRequest(this.uuid != null ^ this.key != null, MSG_ID_OR_KEY_MUST_BE_PROVIDED);
```

Comment:
This line makes line 54 of the same file obsolete. That line should be removed. We do the same check (but written differently) twice.